src/test/java/com/github/princesslana/eriscasper/data/TestData.java

```diff
@@ -0,0 +1,71 @@
+package com.github.princesslana.eriscasper.data;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import org.assertj.core.api.Assertions;
+import org.testng.annotations.Test;
+
+public class TestData {
+
+  @Test
+  public void toJsonString_whenPresent_shouldInclude() throws DataException {
+    Assertions.assertThat(Data.toJsonString(Obj.of(Optional.of("value"))))
+        .isEqualTo("{\"field\":\"value\"}");
+  }
+
+  @Test
+  public void toJsonString_whenNull_shouldInclude() throws DataException {
+    Assertions.assertThat(Data.toJsonString(Obj.of(null))).isEqualTo("{\"field\":null}");
+  }
+
+  @Test
+  public void toJsonString_whenEmpty_shouldOmit() throws DataException {
+    Assertions.assertThat(Data.toJsonString(Obj.of(Optional.empty()))).isEqualTo("{}");
+  }
+
+  @Test
+  public void toJsonString_whenEmptyList_shouldInclude() throws DataException {
+    Assertions.assertThat(Data.toJsonString(Obj.of(Optional.of(Collections.emptyList()))))
+        .isEqualTo("{\"field\":[]}");
+  }
+
+  @Test
+  public void toJsonString_whenPresentInMap_shouldInclude() throws DataException {
+    Assertions.assertThat(Data.toJsonString(mapOf("key", Optional.of("value"))))
+        .isEqualTo("{\"key\":\"value\"}");
+  }
+
+  @Test
+  public void toJsonString_whenNullInMap_shouldInclude() throws DataException {
+    Assertions.assertThat(Data.toJsonString(mapOf("key", null))).isEqualTo("{\"key\":null}");
+  }
+
+  @Test
+  public void toJsonString_whenEmptyInMap_shouldOmit() throws DataException {
+    Assertions.assertThat(Data.toJsonString(mapOf("key", Optional.empty()))).isEqualTo("{}");
+  }
+
+  @Test
+  public void toJsonString_whenEmptyListInMap_shouldInclude() throws DataException {
+    Assertions.assertThat(Data.toJsonString(mapOf("key", Optional.of(Collections.emptyList()))))
+        .isEqualTo("{\"key\":[]}");
+  }
+
+  private static class Obj<T> {
```

Comment:
Is Obj really a good name here? Maybe we could use something more descriptive? Not 100% sure about it though.

Reply:
This is just an internally used testing class.

The name is chosen such that we have `mapOf` and `Obj.of` (and we can't use Object) to have tests that serialize json from a Map, and json from an Object.